### ISSUE TITLE
2.0.4 - 2.1.1 broken on Wine/Linux, 2.0.3 works

### ISSUE THREAD

**Issue by infinity0** [stemrollerapp/stemroller#80]
All songs fail to demux even when I set CPU mode on. Console logs contains:

~~~~
Trace: [Error: ENOENT: no such file or directory, scandir 'C:\users\infinity0\Temp\StemRoller-ohuoIY\separated\htdemucs_ft'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\users\\infinity0\\Temp\\StemRoller-ohuoIY\\separated\\htdemucs_ft'
}
    at processVideo (G:\bin\stemroller-2.0.4-win-cuda\resources\app.asar\main-src\processQueue.cjs:344:13)
~~~~

So, looks like something is wrong with the tempdir handling.

**Comment by iffyloop** [comment]
FYI: it will probably be easier to install Node.js/NPM and Python manually via your package manager, install Demucs globally via `pip`, then run StemRoller in development mode, instead of using Wine. I'm not currently maintaining pre-built binaries for Linux but there's no Windows-specific dependencies that would prevent StemRoller from running on Linux.

**Comment by infinity0** [comment]
Good to know, thanks! I'll give it a go. Do you know if ROCm (amdgpu) support is feasible and would you take a PR for it?

**Comment by iffyloop** [comment]
Yes - it should work pretty much automatically if you install a version of Demucs which is compatible with ROCm! If you run into issues, you can try passing additional arguments into the Demucs process here: `https://github.com/stemrollerapp/stemroller/blob/main/main-src/processQueue.cjs#L302` (I wonder if it needs extra arguments in case it doesn't select the correct GPU? I don't have any CUDA or ROCm device unfortunately, so can't test myself...)

Thanks for offering to create a PR - please let me know if it takes any code changes to get ROCm working, and let's discuss further after that.

**Comment by iffyloop** [comment]
(Select Linux and ROCm at https://pytorch.org/get-started/locally/ to find out how to install PyTorch correctly)

**Comment by iffyloop** [comment]
Closing due to inactivity
